Sample Crunch/App.xaml.cs

```diff
@@ -21,77 +22,30 @@ protected override void OnStartup(System.Windows.StartupEventArgs e)
         {
             // Show splash screen
             SplashScreen splash = new SplashScreen();
-            splash.ShowCloseButton = false;
             splash.Show();
+            splash.Topmost = true;
 
-            // Start a stop watch   
-            Stopwatch timer = new Stopwatch();
-            timer.Start();
+            // Note: This can not be done before splash because types are registered in ViewModelLocator
+            var upd = SimpleIoc.Default.GetInstance<ViewModel.UpdateViewModel>();
 
-            // Check for updates
+            // Check for updates async
             var updateTask = Task.Run(async () =>
             {
-                await UpdateApp();
+                await upd.CheckForUpdates(5000);
             });
 
             // Load your windows but don't show it yet   
             base.OnStartup(e);
-
-            // Create main window
             MainWindow main = new MainWindow();
-            MainWindow = main;
-            this.ShutdownMode = System.Windows.ShutdownMode.OnExplicitShutdown;
-
-            timer.Stop();
 
-            int remainingTimeToShowSplash = MINIMUM_SPLASH_TIME - (int)timer.ElapsedMilliseconds;
-            if (remainingTimeToShowSplash > 0)
-                Thread.Sleep(remainingTimeToShowSplash);
-
-            splash.Close();
-        }
+            updateTask.Wait(2000);
```

Comment:
Don't see the purpose of this since it already has a timeout in CheckForUpdates. The purpose here should be to show the splash long enought as done before.